### ISSUE TITLE
:( Unable to Rebuild Apk with Apktool?

### ISSUE THREAD

**Issue by V3rB0se** [iBotPeaches/Apktool#1974]
unable to rebuild apk>/

**version of apktool** `2.3.4-dirty`
**operating system** `Linux Bleep 4.12.0-kali1-amd64 #1 SMP Debian 4.12.6-1kali6 (2017-08-30) x86_64 GNU/Linux`

`[-] I: **Using Apktool 2.3.4-dirty**
Exception in thread "main" java.lang.NoSuchMethodError: brut.androlib.meta.StringExConstructor.constructScalar(Lorg/yaml/snakeyaml/nodes/ScalarNode;)Ljava/lang/String;
	at brut.androlib.meta.StringExConstructor.access$100(StringExConstructor.java:25)
	at brut.androlib.meta.StringExConstructor$ConstructStringEx.construct(StringExConstructor.java:32)
	at org.yaml.snakeyaml.constructor.Constructor$ConstructScalar.constructStandardJavaInstance(Constructor.java:435)
	at org.yaml.snakeyaml.constructor.Constructor$ConstructScalar.construct(Constructor.java:382)
	at org.yaml.snakeyaml.constructor.BaseConstructor.constructObject(BaseConstructor.java:182)
	at org.yaml.snakeyaml.constructor.Constructor$ConstructMapping.constructJavaBean2ndStep(Constructor.java:239)
	at org.yaml.snakeyaml.constructor.Constructor$ConstructMapping.construct(Constructor.java:190)
	at org.yaml.snakeyaml.constructor.Constructor$ConstructYamlObject.construct(Constructor.java:346)
	at org.yaml.snakeyaml.constructor.BaseConstructor.constructObject(BaseConstructor.java:182)
	at org.yaml.snakeyaml.constructor.BaseConstructor.constructDocument(BaseConstructor.java:141)
	at org.yaml.snakeyaml.constructor.BaseConstructor.getSingleData(BaseConstructor.java:127)
	at org.yaml.snakeyaml.Yaml.loadFromReader(Yaml.java:450)
	at org.yaml.snakeyaml.Yaml.loadAs(Yaml.java:444)
	at brut.androlib.meta.MetaInfo.load(MetaInfo.java:70)
	at brut.androlib.Androlib.readMetaFile(Androlib.java:262)
	at brut.androlib.Androlib.build(Androlib.java:276)
	at brut.androlib.Androlib.build(Androlib.java:269)
	at brut.apktool.Main.cmdBuild(Main.java:238)
	at brut.apktool.Main.main(Main.java:76)
Error: Unable to rebuild apk with apktool`

**Comment by dsx12** [comment]
This is a active bug with no current solution I myself have been searching and asking questions for days.

The developer of the apktool @iBotPeaches advised me to use the aapt2 parameter (--use-aapt2).

I am waiting for him to respond and help me understand that parameter and how to use it a bit more @ https://github.com/iBotPeaches/Apktool/issues/1934#issuecomment-451061787

Other than that I have attempted to use it on windows to no avail it seems that weaponizing apks is on hold at the moment until we are able to fully understand why?

The script is decompiling the apks fine but as for recompiling the decompiled there is not current method that is working.

You would have more luck building your own android application and injecting the payload into it. :/

Kali and other penetration testing/security applications can't update metasploit frameworks "msfvenom -x" function until apktool fixes the issue so until then we have to wait on @iBotPeaches to provide a solution or update the source code of apktool.

**Comment by V3rB0se** [comment]
> This is a active bug with no current solution I myself have been searching and asking questions for days.
> 
> The developer of the apktool @iBotPeaches advised me to use the aapt2 parameter (--use-aapt2).
> 
> I am waiting for him to respond and help me understand that parameter and how to use it a bit more @ [#1934 (comment)](https://github.com/iBotPeaches/Apktool/issues/1934#issuecomment-451061787)
> 
> Other than that I have attempted to use it on windows to no avail it seems that weaponizing apks is on hold at the moment until we are able to fully understand why?
> 
> The script is decompiling the apks fine but as for recompiling the decompiled code there is not a current method that is working.
> 
> You would have more luck building your own android application and injecting the payload into it. :/
> 
> Kali and other penetration testing/security operating systems can't update metasploit framework's "msfvenom -x" function until apktool fixes the issue so until then we have to wait on @iBotPeaches to provide a solution or update the source code of apktool.
> 
> Besides building android applications this tools second most used feature is the decompiling and recompiling android applications for security or other reasons. So I imagine it shouldn't be long until apktool is updated.

                                                           **Lets Do It Manually** 
[](https://null-byte.wonderhowto.com/how-to/embed-metasploit-payload-original-apk-file-part-2-do-manually-0167124/)

**Comment by V3rB0se** [comment]
**Lets Do It Manually** `https://null-byte.wonderhowto.com/how-to/embed-metasploit-payload-original-apk-file-part-2-do-manually-0167124/`

**Comment by V3rB0se** [comment]
Lets Do It Manually [Manually Embed Payload](https://null-byte.wonderhowto.com/how-to/embed-metasploit-payload-original-apk-file-part-2-do-manually-0167124/)

**Comment by V3rB0se** [comment]
[Lets Do It Manually](https://null-byte.wonderhowto.com/how-to/embed-metasploit-payload-original-apk-file-part-2-do-manually-0167124/)

**Comment by dsx12** [comment]
That method doesn't work if apktool cannot rebuild the apk did it work for you? I tried them already and they failed.

**Comment by iBotPeaches** [comment]
Couldn't replicate this issue with v2.3.4. Since the stacktrace says "dirty", you must have built it yourself. Please ensure you follow build instructions properly.

**Comment by dsx12** [comment]
Your last comment saying "I don't think reverse engineering is for you" why make this comment because I am asking questions? Not everyone who uses your tool is a expert I was simply asking for a example because clearly It's not only me asking for the answer man... It came off as a bit rude to say that @iBotPeaches .. There are many people asking how for a fix or example and recompile apks with apktool. You can't expect everyone to be a expert who uses your program..

**Comment by iBotPeaches** [comment]
There is a time and place for support @dsx12, unforunately a tool bug tracker is not it. GitHub may have changed that stigma, but a bug tracker is for issues and sometimes features in a tool. I get an email for every response here and that just begins to scratch the surface for every PM, email, etc I get for Apktool.

The tool has many problems, so I cannot fit my time with support in there as well. There are forums, IRC chats, Stackoverflow and more for helping you understand things.

While the comment may have been rudely worded, I recommend you just take a step back and look at the problem. Do some research and read around online. There are plenty of things I don't know and I don't head straight for the owner of said package and submit a bug report. I took plenty of time out of my day to properly respond to those claims and researched them. I told you what to do, but I can't sit here and leave a highly detailed step by step guide. Gotta meet me halfway here.

I wish you luck, but maintainers unforunately don't have the time to sit down and walk people through every little step.

**Comment by V3rB0se** [comment]
@dsx12 on hangouts?

**Comment by ssnkhan** [comment]
@V3rB0se Thanks - this solved the issue! Definitely could do highlighting this trivial fix more prominently given the corpus of Kali users who are posting about this.

**Comment by V3rB0se** [comment]
Shukria Janab

**Comment by V3rB0se** [comment]
Koi Baat Nahin 😜

**Comment by azamet90** [comment]
> ing Kali to Its Latest Version.

i did but result is same manually injecting is same ...
@V3rB0se please try on this app yourself and see: [Download](http://d1.dl-android.com/apps/facebook-mod[dlandroid.com].apk)
